sunpy/coordinates/frames.py

```diff
@@ -847,3 +858,186 @@ class GeocentricEarthEquatorial(SunPyBaseCoordinateFrame):
     Aberration due to Earth motion is not included.
     """
     equinox = TimeFrameAttributeSunPy(default=_J2000)
+
+
+class BaseMagnetic(SunPyBaseCoordinateFrame):
+    """
+    Base class for frames that rely on the Earth's magnetic model (MAG, SM, and GSM).
+
+    This class is not intended to be used directly and has no transformations defined.
+    """
+    magnetic_model = Attribute(default='igrf13')
+
+    @property
+    def _igrf_file(self):
+        if not self.magnetic_model.startswith("igrf"):
+            raise ValueError
+
+        # First look if the file is bundled in package
+        local_file = os.path.join(os.path.dirname(__file__), "data",
+                                  f"{self.magnetic_model}coeffs.txt")
+        if os.path.exists(local_file):
+            return local_file
+
+        # Otherwise download the file and cache it
+        return cache.download("https://www.ngdc.noaa.gov/IAGA/vmod/coeffs/"
+                              f"{self.magnetic_model}coeffs.txt", expiry=None)
```

Comment:
I've switched back to using Astropy's `download_file()` instead of our `cache.download()`.  I don't like the expiry behavior of our cache for these IGRF coefficient files